feature/experimental/isis/ate_tests/isis_interface_hello_padding_enable_test/isis_interface_hello_padding_enable_test.go

```diff
@@ -0,0 +1,460 @@
+		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
+		WithHeaders(ondatra.NewEthernetHeader(), v4Header)
+
+	t.Log("Configuring v6 traffic flow ")
+	v6Header := ondatra.NewIPv6Header()
+	v6Header.DstAddressRange().WithMin(v6IP).WithCount(1)
+
+	v6Flow := ate.Traffic().NewFlow("v6Flow").
+		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
+		WithHeaders(ondatra.NewEthernetHeader(), v6Header)
+
+	return []*ondatra.Flow{v4Flow, v6Flow}
+}
+
+// TestIsisInterfaceHelloPaddingEnable verifies adjacency with hello padding enabled.
+func TestIsisInterfaceHelloPaddingEnable(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	intfName := dut.Port(t, "port1").Name()
+
+	// Configure interface on the DUT.
+	configureDUT(t)
+
+	// Configure network Instance type on DUT.
+	dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+	t.Log("Pushing network Instance type config on DUT")
+	gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+
+	// Configure isis on DUT.
+	configureISIS(t, dut, intfName)
+
+	// Configure interface,isis and traffic on ATE.
+	ateTopo := configureATE(t, ate)
+	flows := createFlows(t, ate, ateTopo)
+	port1 := ate.Port(t, "port1")
+
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		intfName = intfName + ".0"
+	}
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, isisInstance).Isis()
+
+	t.Run("Isis telemetry", func(t *testing.T) {
+
+		adjacencyPath := statePath.Interface(intfName).Level(2).AdjacencyAny().AdjacencyState().State()
+
+		_, ok := gnmi.WatchAll(t, dut, adjacencyPath, time.Minute, func(val *ygnmi.Value[oc.E_Isis_IsisInterfaceAdjState]) bool {
+			state, present := val.Val()
+			return present && state == oc.Isis_IsisInterfaceAdjState_UP
+		}).Await(t)
+		if !ok {
+			t.Fatalf("No isis adjacency reported on interface %v", intfName)
+		}
+		// Getting neighbors sysid.
+		sysid := gnmi.GetAll(t, dut, statePath.Interface(intfName).Level(2).AdjacencyAny().SystemId().State())
+		ateSysID := sysid[0]
+		ateLspID := ateSysID + ".00-00"
+
+		t.Run("HelloPadding checks", func(t *testing.T) {
+			if got := gnmi.Get(t, dut, statePath.Global().HelloPadding().State()); got != oc.Isis_HelloPaddingType_ADAPTIVE {
+				t.Errorf("FAIL- Expected global hello padding state not found, got %d, want %d", got, oc.Isis_HelloPaddingType_ADAPTIVE)
+			}
+			if got := gnmi.Get(t, dut, statePath.Interface(intfName).HelloPadding().State()); got != oc.Isis_HelloPaddingType_ADAPTIVE {
+				t.Errorf("FAIL- Expected interface hello padding state not found, got %d, want %d", got, oc.Isis_HelloPaddingType_ADAPTIVE)
+			}
+
+			// Changing MTU at ATE side
+			ateTopo.Interfaces()[atePort1attr.Name].WithPort(port1).Ethernet().WithMTU(2000)
+			ateTopo.Push(t).StartProtocols(t)
+
+			// Adjacency check
+			time.Sleep(time.Second * 20)
```

Comment:
Can you replace the sleep + gnmi.Lookup into a single gnmi.Watch call?

Reply:
fixed in latest commit